File: memory-domain/src/main/java/com/memory/dto/search/MigrationResponse.java
package com.memory.dto.search;

import lombok.Getter;

@Getter
public class MigrationResponse {
    
    private final boolean success;
    private final String message;
    private final long processedCount;
    private final long successCount;
    private final long errorCount;
    private final long elapsedTimeMs;

    public MigrationResponse(boolean success, String message, long processedCount, long successCount, long errorCount, long elapsedTimeMs) {
        this.success = success;
        this.message = message;
        this.processedCount = processedCount;
        this.successCount = successCount;
        this.errorCount = errorCount;
        this.elapsedTimeMs = elapsedTimeMs;
    }

    public static MigrationResponse success(String message, long processedCount,
                                            long successCount, long errorCount, long elapsedTimeMs) {
        return new MigrationResponse(true, message, processedCount, successCount, errorCount, elapsedTimeMs);
    }
    
    public static MigrationResponse failure(String message, long processedCount, 
                                          long successCount, long errorCount, long elapsedTimeMs) {
        return new MigrationResponse(false, message, processedCount, successCount, errorCount, elapsedTimeMs);
    }
}

File: memory-domain/src/main/kotlin/com/memory/dto/search/MigrationResponse.kt
package com.memory.dto.search

data class MigrationResponse(
    val success: Boolean,
    val message: String,
    val processedCount: Long,
    val successCount: Long,
    val errorCount: Long,
    val elapsedTimeMs: Long
) {
    companion object {
        @JvmStatic
        fun success(
            message: String,
            processedCount: Long,
            successCount: Long,
            errorCount: Long,
            elapsedTimeMs: Long
        ): MigrationResponse {
            return MigrationResponse(
                success = true,
                message = message,
                processedCount = processedCount,
                successCount = successCount,
                errorCount = errorCount,
                elapsedTimeMs = elapsedTimeMs
            )
        }

        @JvmStatic
        fun failure(
            message: String,
            processedCount: Long,
            successCount: Long,
            errorCount: Long,
            elapsedTimeMs: Long
        ): MigrationResponse {
            return MigrationResponse(
                success = false,
                message = message,
                processedCount = processedCount,
                successCount = successCount,
                errorCount = errorCount,
                elapsedTimeMs = elapsedTimeMs
            )
        }
    }
}
